### ISSUE TITLE
[wasm] [debugger] `MonoProxy` behaves incorrectly when value for the `columnNumber` property is missing in `Debugger.setBreakpointByUrl` call

### ISSUE THREAD

**Issue by seclerp** [dotnet/runtime#81321]
### Description

When sending a `setBreakpointByUrl` request without providing a column (a case when we want to implement "line breakpoint"), `BlazorDebugProxy` will always fail to resolve `location`. This is because validation done against location checks the value of the `columnNumber` - when its value equals `null` it will just silently ignore provided `location` and return a non-resolved breakpoint with empty `locations`.

WIP/CDP [declares](https://chromedevtools.github.io/devtools-protocol/tot/Debugger/#method-setBreakpointByUrl) `column` as optional and allows to not passing it to achieve success resolving for such breakpoint set commands.

I'm interested in creating PR with a fix if it works.

### Reproduction Steps

Send such a request to proxy:
```json
{
  "id": 12,
  "method": "Debugger.setBreakpointByUrl",
  "params": {
    "lineNumber": 14,
    "urlRegex": "[/\\\\][cC][oO][uU][nN][tT][eE][rR]\\.[rR][aA][zZ][oO][rR]([;?#!].*)?$"
  }
}
```

There are no specific scenarios, it will reproduce inside any method.

### Expected behavior

The expected response should include resolved locations, like below:

```json
{
  "id": 12,
  "result": {
    "breakpointId": "2:14:0:[/\\\\][cC][oO][uU][nN][tT][eE][rR]\\.[rR][aA][zZ][oO][rR]([;?#!].*)?$",
    "locations": [
      {
        "scriptId": "dotnet://191_4",
        "lineNumber": 14,
        "columnNumber": 4
      }
    ]
  }
}
``` 

### Actual behavior

Currently proxy returns empty `locations`:
```json
{
  "id": 12,
  "result": {
    "breakpointId": "2:14:0:[/\\\\][cC][oO][uU][nN][tT][eE][rR]\\.[rR][aA][zZ][oO][rR]([;?#!].*)?$",
    "locations": []
  }
}
```

### Known Workarounds

The command succeeds when sending `0` as `columnNumber`.

### Configuration

```
.NET SDK:
 Version:   7.0.100
 Commit:    e12b7af219

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  13.1
 OS Platform: Darwin
 RID:         osx.13-arm64
 Base Path:   /usr/local/share/dotnet/sdk/7.0.100/

Host:
  Version:      7.0.0
  Architecture: arm64
  Commit:       d099f075e4

.NET SDKs installed:
  6.0.404 [/usr/local/share/dotnet/sdk]
  7.0.100 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.12 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.12 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  x64   [/usr/local/share/dotnet/x64]
    registered at [/etc/dotnet/install_location_x64]
```

### Other information

Let me provide an example of how it works for JS code in Chrome:

Request:
```json
  {
    "id": 40,
    "method": "Debugger.setBreakpointByUrl",
    "params": {
      "lineNumber": 13,
      "urlRegex": "^http://[lL][oO][cC][aA][lL][hH][oO][sS][tT]:63344/([sS][iI][mM][pP][lL][eE][bB][lL][aA][zZ][oO][rR][aA][pP][pP][nN][eE][tT]7/[sS][iI][mM][pP][lL][eE][bB][lL][aA][zZ][oO][rR][aA][pP][pP][nN][eE][tT]7/|[^.;?#]+)?([iI][nN][dD][eE][xX]\\.[hH][tT][mM][lL])?([;?#!].*)?$"
    }
  }
```

Response:
```json
  {
    "id": 40,
    "result": {
      "breakpointId": "2:13:0:^http://[lL][oO][cC][aA][lL][hH][oO][sS][tT]:63344/([sS][iI][mM][pP][lL][eE][bB][lL][aA][zZ][oO][rR][aA][pP][pP][nN][eE][tT]7/[sS][iI][mM][pP][lL][eE][bB][lL][aA][zZ][oO][rR][aA][pP][pP][nN][eE][tT]7/|[^.;?#]+)?([iI][nN][dD][eE][xX]\\.[hH][tT][mM][lL])?([;?#!].*)?$",
      "locations": [
        {
          "scriptId": "11",
          "lineNumber": 13,
          "columnNumber": 12
        }
      ]
    }
  }
```

**Comment by lewing** [comment]
iirc It should do what you expect if you set column to zero

**Comment by seclerp** [comment]
Yes, that's the workaround I'm using. But I think it's worth fixing because such incorrect behavior breaks the semantics of `columnNumber`, which is optional.

**Comment by lewing** [comment]
fixed in #81336
